TYPO3.Neos/Classes/TYPO3/Neos/Service/View/NodeView.php

```diff
@@ -237,7 +237,9 @@ public function collectParentNodeData(NodeInterface $rootNode, array $nodes)
                     $collectTreeNodeData($children, $childNode);
                 }
             }
-            $treeNodes[] = $self->collectTreeNodeData($node['node'], true, $children, $children !== array(), isset($node['matched']));
+            if (isset($node['node']) && $node['node'] instanceof NodeInterface) {
```

Comment:
Running node:repair on +100'000 nodes project is bit hard (not enough memory on the server). And I think the system should be solid enough to run even if we have some orphan nodes in the the CR, will try to do a bit more debugging.